### PR TITLE
adds new release CR

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/3313.

### CODE REVIEW

File: pkg/apis/core/v1alpha1/register.go [hunk 1]
```diff
@@ -35,6 +35,8 @@ var knownTypes = []runtime.Object{
 	&IngressConfigList{},
 	&NodeConfig{},
 	&NodeConfigList{},
+	&Release{},
+	&ReleaseList{},
```
Comment (xh3b4sd) [inline pkg/apis/core/v1alpha1/register.go]: The naming is cleaner as first suggested by @oponder.

File: pkg/apis/core/v1alpha1/release_types.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+type ReleaseSpec struct {
+	Operator      OperatorSpecGuest        `json:"operator" yaml:"operator"`
+	VersionBundle ReleaseSpecVersionBundle `json:"versionBundle" yaml:"versionBundle"`
+}
+
+type OperatorSpecGuest struct {
+	Name    string `json:"cluster" yaml:"name"`
+	Version string `json:"node" yaml:"version"`
+}
```
Comment (xh3b4sd) [inline pkg/apis/core/v1alpha1/release_types.go]: There is only name and version relevant. `customResource.Spec.Operator.Name` reads well.

File: pkg/apis/core/v1alpha1/release_types.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	Conditions []ReleaseStatusCondition `json:"conditions" yaml:"conditions"`
+}
+
+// ReleaseStatusCondition expresses a condition in which a node may is.
+type ReleaseStatusCondition struct {
+	// Status may be True, False or Unknown.
+	Status string `json:"status" yaml:"status"`
+	// Type may be Pending, Ready, Draining, Drained.
+	Type string `json:"type" yaml:"type"`
+}
```
Comment (xh3b4sd) [inline pkg/apis/core/v1alpha1/release_types.go]: Having the usual status support.

File: pkg/apis/core/v1alpha1/zz_generated.deepcopy.go [hunk 4]
```diff
@@ -1834,6 +1834,154 @@ func (in *NodeConfigStatusCondition) DeepCopy() *NodeConfigStatusCondition {
 	return out
 }
 
+// DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
+func (in *OperatorSpecGuest) DeepCopyInto(out *OperatorSpecGuest) {
```
Comment (xh3b4sd) [inline pkg/apis/core/v1alpha1/zz_generated.deepcopy.go]: All the generated crap.

File: pkg/apis/core/v1alpha1/release_types.go [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+	Spec              ReleaseSpec   `json:"spec"`
+	Status            ReleaseStatus `json:"status"`
+}
+
+type ReleaseSpec struct {
+	Operator      OperatorSpecGuest        `json:"operator" yaml:"operator"`
+	VersionBundle ReleaseSpecVersionBundle `json:"versionBundle" yaml:"versionBundle"`
+}
+
+type OperatorSpecGuest struct {
```
Comment (kopiczko) [inline pkg/apis/core/v1alpha1/release_types.go]: Shouldn't it be `ReleaseSpecOperator`?

Comment (xh3b4sd) [reply pkg/apis/core/v1alpha1/release_types.go]: For sure. Fixed.

**Comment by xh3b4sd** [comment]
PTALA

**Review by MarcelMue — APPROVED** [review]
LGTM

**Comment by xh3b4sd** [comment]
FYI the status magic wasn't right. Fixed. Going when green.

**Pull request merged**